shared/feature/feeding/src/main/java/com/epmedu/animeal/feeding/presentation/viewmodel/FeedingConfirmationState.kt

```diff
@@ -1,5 +1,13 @@
 package com.epmedu.animeal.feeding.presentation.viewmodel
 
-enum class FeedingConfirmationState {
-    Dismissed, Loading, Showing, FeedingStarted
+import com.epmedu.animeal.common.constants.DefaultConstants.EMPTY_STRING
+
+interface FeedingConfirmationState {
+    companion object {
+        const val REQUEST_CANCELLED = "RequestCancelled"
+    }
+    data class Dismissed(val reason: String = EMPTY_STRING) : FeedingConfirmationState
```

Comment:
Can we return Dismissed back to the object as it was? Maybe, 'Dismissed' has not the best naming, yet from my understanding, it should be treated as "Clean", initial state of FeedingConfirmationState, without expecting any additional logic behind it. Plus, "reason", basically, makes it act as 2 different states, depending on which string was inserted.
My suggestion :
Add a new state for FeedingConfrimationState, named smth like "FeedingReservationError" or "ShowingErrorDialog". This state can take a string as an error message, but that won't be REQUEST_CANCELLED string, but the message of the dialog itself - what exactly we want to show the user. On this FeedingReservationError state we show AlertDialog, and "onDismiss" of the dialog we return our state back to 'object Dismissed'. That would make this new state universal for all next possible errors, just showing different messages for each

Reply:
Oups, forgot we are sending this error state from viewModel which can't access string resources. Let's, then, make new state as object and name it specifically for this issue, smth like "FeedingIsAlreadyBooked". In the future, if we have the need to show different error messages, we can separate possible errors in a separate interface, like:

`interface FeedingConfirmationState {
    
    object Dismissed : FeedingConfirmationState
    object Loading : FeedingConfirmationState
    object Showing : FeedingConfirmationState
    object FeedingStarted : FeedingConfirmationState

    sealed interface FeedingConfirmationError: FeedingConfirmationState {
        object FeedingWasAlreadyBooked: FeedingConfirmationError
    }
}`